### ISSUE TITLE
unexpected debugging error with events

### ISSUE THREAD

**Issue by aspark21** [brickfield/moodle-mod_board#104]
Moodle 4.2
PHP8.0

```
mod_board.board_test/test_board_add_column
mod_board\board_test::test_board_add_column
Unexpected debugging() call detected.
Debugging: Invalid event functions defined in \mod_board\event\add_column
* line 275 of /lib/classes/event/base.php: call to debugging()
* line 465 of /mod/board/classes/board.php: call to core\event\base::create()
* line 447 of /mod/board/classes/board.php: call to mod_board\board::board_add_column_log()
* line 190 of /mod/board/tests/board_test.php: call to mod_board\board::board_add_column()
* line 1548 of /vendor/phpunit/phpunit/src/Framework/TestCase.php: call to mod_board\board_test->test_board_add_column()
* line 1154 of /vendor/phpunit/phpunit/src/Framework/TestCase.php: call to PHPUnit\Framework\TestCase->runTest()
* line 81 of /lib/phpunit/classes/advanced_testcase.php: call to PHPUnit\Framework\TestCase->runBare()
* line 728 of /vendor/phpunit/phpunit/src/Framework/TestResult.php: call to advanced_testcase->runBare()
* line 904 of /vendor/phpunit/phpunit/src/Framework/TestCase.php: call to PHPUnit\Framework\TestResult->run()
* line 675 of /vendor/phpunit/phpunit/src/Framework/TestSuite.php: call to PHPUnit\Framework\TestCase->run()
* line 675 of /vendor/phpunit/phpunit/src/Framework/TestSuite.php: call to PHPUnit\Framework\TestSuite->run()
* line 675 of /vendor/phpunit/phpunit/src/Framework/TestSuite.php: call to PHPUnit\Framework\TestSuite->run()
* line 653 of /vendor/phpunit/phpunit/src/TextUI/TestRunner.php: call to PHPUnit\Framework\TestSuite->run()
* line 144 of /vendor/phpunit/phpunit/src/TextUI/Command.php: call to PHPUnit\TextUI\TestRunner->run()
* line 97 of /vendor/phpunit/phpunit/src/TextUI/Command.php: call to PHPUnit\TextUI\Command->run()
* line 98 of /vendor/phpunit/phpunit/phpunit: call to PHPUnit\TextUI\Command::main()
* line 122 of /vendor/bin/phpunit: call to include()

mod_board.board_test/test_board_update_column
mod_board\board_test::test_board_update_column
Unexpected debugging() call detected.
Debugging: Invalid event functions defined in \mod_board\event\update_column
* line 275 of /lib/classes/event/base.php: call to debugging()
* line 516 of /mod/board/classes/board.php: call to core\event\base::create()
* line 494 of /mod/board/classes/board.php: call to mod_board\board::board_update_column_log()
* line 204 of /mod/board/tests/board_test.php: call to mod_board\board::board_update_column()
* line 1548 of /vendor/phpunit/phpunit/src/Framework/TestCase.php: call to mod_board\board_test->test_board_update_column()
* line 1154 of /vendor/phpunit/phpunit/src/Framework/TestCase.php: call to PHPUnit\Framework\TestCase->runTest()
* line 81 of /lib/phpunit/classes/advanced_testcase.php: call to PHPUnit\Framework\TestCase->runBare()
* line 728 of /vendor/phpunit/phpunit/src/Framework/TestResult.php: call to advanced_testcase->runBare()
* line 904 of /vendor/phpunit/phpunit/src/Framework/TestCase.php: call to PHPUnit\Framework\TestResult->run()
* line 675 of /vendor/phpunit/phpunit/src/Framework/TestSuite.php: call to PHPUnit\Framework\TestCase->run()
* line 675 of /vendor/phpunit/phpunit/src/Framework/TestSuite.php: call to PHPUnit\Framework\TestSuite->run()
* line 675 of /vendor/phpunit/phpunit/src/Framework/TestSuite.php: call to PHPUnit\Framework\TestSuite->run()
* line 653 of /vendor/phpunit/phpunit/src/TextUI/TestRunner.php: call to PHPUnit\Framework\TestSuite->run()
* line 144 of /vendor/phpunit/phpunit/src/TextUI/Command.php: call to PHPUnit\TextUI\TestRunner->run()
* line 97 of /vendor/phpunit/phpunit/src/TextUI/Command.php: call to PHPUnit\TextUI\Command->run()
* line 98 of /vendor/phpunit/phpunit/phpunit: call to PHPUnit\TextUI\Command::main()
* line 122 of /vendor/bin/phpunit: call to include()


/var/www/html/lib/phpunit/classes/advanced_testcase.php:89

mod_board.board_test/test_board_delete_column
mod_board\board_test::test_board_delete_column
Unexpected debugging() call detected.
Debugging: Invalid event functions defined in \mod_board\event\delete_column
* line 275 of /lib/classes/event/base.php: call to debugging()
* line 588 of /mod/board/classes/board.php: call to core\event\base::create()
* line 548 of /mod/board/classes/board.php: call to mod_board\board::board_delete_column_log()
* line 216 of /mod/board/tests/board_test.php: call to mod_board\board::board_delete_column()
* line 1548 of /vendor/phpunit/phpunit/src/Framework/TestCase.php: call to mod_board\board_test->test_board_delete_column()
* line 1154 of /vendor/phpunit/phpunit/src/Framework/TestCase.php: call to PHPUnit\Framework\TestCase->runTest()
* line 81 of /lib/phpunit/classes/advanced_testcase.php: call to PHPUnit\Framework\TestCase->runBare()
* line 728 of /vendor/phpunit/phpunit/src/Framework/TestResult.php: call to advanced_testcase->runBare()
* line 904 of /vendor/phpunit/phpunit/src/Framework/TestCase.php: call to PHPUnit\Framework\TestResult->run()
* line 675 of /vendor/phpunit/phpunit/src/Framework/TestSuite.php: call to PHPUnit\Framework\TestCase->run()
* line 675 of /vendor/phpunit/phpunit/src/Framework/TestSuite.php: call to PHPUnit\Framework\TestSuite->run()
* line 675 of /vendor/phpunit/phpunit/src/Framework/TestSuite.php: call to PHPUnit\Framework\TestSuite->run()
* line 653 of /vendor/phpunit/phpunit/src/TextUI/TestRunner.php: call to PHPUnit\Framework\TestSuite->run()
* line 144 of /vendor/phpunit/phpunit/src/TextUI/Command.php: call to PHPUnit\TextUI\TestRunner->run()
* line 97 of /vendor/phpunit/phpunit/src/TextUI/Command.php: call to PHPUnit\TextUI\Command->run()
* line 98 of /vendor/phpunit/phpunit/phpunit: call to PHPUnit\TextUI\Command::main()
* line 122 of /vendor/bin/phpunit: call to include()


/var/www/html/lib/phpunit/classes/advanced_testcase.php:89
```

**Comment by learningtechnologyservices** [comment]
Hi Alistair, yes, thanks for reporting this, this bug is on our TODO list, as the original event names don't comply with the [Moodle event naming conventions here](https://docs.moodle.org/dev/Events_API#Events_naming_convention).

**Comment by learningtechnologyservices** [comment]
Hi Alistair, thanks, this has been fixed with [commit fb53f6c](https://github.com/brickfield/moodle-mod_board/commit/fb53f6c0096d152c34d9c033abe4f0f48417d098).
